examples/esp_idf/rpc/README.md

```diff
@@ -0,0 +1,54 @@
+# RPC Example
+
+## Overview
+
+This sample application demonstrates how to use Remote Procedure calls
+with Golioth to issue commands to the device and receive a status
+response and optional data payload in return.
+
+## Requirements
+
+* Golioth credentials
+* Network connectivity
+
+## Building and Running
+
+### Command-line
+
+First, setup the environment. This step assumes you've installed esp-idf
+to `~/esp/esp-idf`. If you haven't, follow the initial steps in
+examples/esp_idf/README.md
+
+```sh
+source ~/esp/esp-idf/export.sh
+```
+You may have to set target based on the ESP32 chip you are using.
+For example, if you are using ESP32-C3, enter this:
+
+```
+idf.py set-target esp32c3
+```
+
+Next, `cd` to the hello example where you can build/flash/monitor:
+
+```
+cd examples/esp_idf/rpc
+idf.py build
+idf.py flash
+idf.py monitor
+```
+## Adding credentials
+
+After building and flashing this app, you will need to add WiFi and Golioth
+credentials using the device shell.
+
+### Shell Credentials
+
+```console
+settings set wifi/ssid "YourWiFiAccessPointName"
+settings set wifi/psk "YourWiFiPassword"
+settings set golioth/psk-id "YourGoliothDevicePSK-ID"
+settings set golioth/psk "YourGoliothDevicePSK"
+```
+
+Type `reset` to restart the app with the new credentials.
```

Comment:
Let's also include some information in the README about how to use this example, similar to the Zephyr version: https://github.com/golioth/golioth-firmware-sdk/blob/main/examples/zephyr/rpc/README.md#issue-an-rpc